src/libraries/System.Transactions.Local/src/System/Transactions/TransactionManager.cs

```diff
@@ -334,6 +359,27 @@ public static TimeSpan MaximumTimeout
 
                 return s_maximumTimeout;
             }
+            set
+            {
+                TransactionsEtwProvider etwLog = TransactionsEtwProvider.Log;
+                if (etwLog.IsEnabled())
+                {
+                    etwLog.MethodEnter(TraceSourceType.TraceSourceBase, "TransactionManager.set_DefaultMaximumTimeout");
+                }
+
+                if(value < TimeSpan.Zero)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
+
+                s_cachedMaxTimeout = false;
+                LazyInitializer.EnsureInitialized(ref s_maximumTimeout, ref s_cachedMaxTimeout, ref s_classSyncObject, () => value);
```

Comment:
Is there value to using the LazyInitializer pattern here? We already have the value. It's just a simple assignment at this point. Why don't we set s_cachedMaxTimeout to true and just assign the value here?

Reply:
I was thinking the same thing. The LazyInitializer handles lazily assigning an object to s_classSyncObject for locking so that might be null. I don't see any reason to need a lock here though.

Reply:
Thanks for the explanation! It's updated to set s_cachedMaxTimeout to true and assign the value directly.